packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -94,7 +105,15 @@ class ModalBarrier extends StatelessWidget {
     final bool modalBarrierSemanticsDismissible = barrierSemanticsDismissible ?? semanticsDismissible;
 
     void handleDismiss() {
-      Navigator.maybePop(context);
+      if (dismissible) {
+        if (onDismiss != null) {
+          onDismiss!();
+        } else {
+          Navigator.maybePop(context);
+        }
```

Comment:
```suggestion
        (onDismiss ?? Navigator.maybePop)();
```

Reply:
As we discussed in chat, this unfortunately won't work as `onDismiss` has a void return type which throws off the ?? evaluation. Bummer, because this was more concise.